### PR TITLE
provide a way to get CSV files of translations in and out of the system

### PR DESCRIPTION
**Pull request by ddellacosta**
https://www.pivotaltracker.com/story/show/158606141

Note that the `resources/dictionary.edn` file I've committed represents the output of first converting the original dictionary data to a CSV using the `edn->csv` function in `gttp2.dev.translations`, adding some translations (just added Japanese so we can either remove those entirely when we get a translator, or whatever), and converting that back into the `resources/dictionary.edn` file using `csv->edn`. Which is just to say that this has been manually tested.

### CODE REVIEW

File: dev-src/gttp2/dev/translations.clj [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+;; Fill this out with all the languages we need
+(def languages
+  {:en "English"
+   :es "Spanish"
+   :ja "Japanese"})
+
+;;
+;; not really correct as we could blow the stack if we recurse
+;; without using recur, but this is so much more clear and I don't
+;; think we'll ever recurse past two or three levels. Also I'm lazy
```
Comment (tank157) [inline dev-src/gttp2/dev/translations.clj]: I totally understand the pain of knowing there's a _potential_ future bug here, but if we've ever nested to the level that this is a problem we've done something very very wrong.

Comment (ddellacosta) [reply dev-src/gttp2/dev/translations.clj]: Yeah, that was basically the reason I sort of stopped fiddling with it and just dropped this comment in. ¯\_(ツ)_/¯

File: src/cljs/gttp2/translations.cljs [hunk 2]
```diff
@@ -57,117 +59,11 @@
...
-      :election-info-timing "If your address is correct and you still get this message, we may not yet have information about your next election. Election information should be available here two to four weeks before your next election - check back here a little closer to election day."}}
-
-    :es
-    {:missing "Missing :es translation!"
-     :language-name "Español"
-
-     :title "Herramienta de Información de Votación"
-
-     :date-spelled (spelled-date "es")}}))
+  (atom (-> (read-dict)
```
Comment (tank157) [inline src/cljs/gttp2/translations.cljs]: I was wondering how you would handle the formatting functions. I'm not super in love with this, but so long as we keep the count to a fairly reasonable level I suppose it's all right. Maybe sometime later we could implement them as a deep merge of maps or something similar.

Comment (ddellacosta) [reply src/cljs/gttp2/translations.cljs]: Yeah this is the main thing I was also not super hot on tbh. Totally receptive to other ideas if you have any!

**Comment by ddellacosta** [comment]
> I might suggest adding some aliases to project.xml that can facilitate the translations in both directions

Good call, will do.

File: src/cljs/gttp2/translations.cljs (hunk 2 above)
Comment (tank157) [reply src/cljs/gttp2/translations.cljs]: It's fine for now, but when we've added our 3rd or 4th language I'll be itching to have the formatters stored something like `{:en {:date-spelled (spelled-date "en")} :es {:date-spelled (spelled-date "es")}}`

Comment (ddellacosta) [reply src/cljs/gttp2/translations.cljs]: Yeah totally...let me review and maybe take another crack at it. I think if we were able to add this stuff directly in the `resources/dictionary.edn` file that'd be handy, assuming there's no issues with converting the edn to Clojure data. I'll also probably want to pretty-print the `dictionary.edn` data so it's not a big blob.

**Comment by tank157** [comment]
Almost forgot the obligatory random emoji ☯️

**Comment by ddellacosta** [comment]
@tank157 Going to merge this in for now as my last experiment wasn't great--will poke at it again later.

**Pull request merged**